median/report_codec.go

```diff
@@ -0,0 +1,49 @@
+package median
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+const typeName = "MedianReport"
+
+func newReportCodec(codec types.Codec) (median.ReportCodec, error) {
+	if codec == nil {
+		return nil, errors.New("codec cannot be nil")
+	}
+	return &reportCodec{codec: codec}, nil
+}
+
+type reportCodec struct {
+	codec types.Codec
+}
+
+var _ median.ReportCodec = reportCodec{}
+
+func (r reportCodec) BuildReport(observations []median.ParsedAttributedObservation) (ocrtypes.Report, error) {
+	if len(observations) == 0 {
+		return nil, fmt.Errorf("cannot build report from empty attributed observations")
+	}
+
+	agg := aggregate(observations)
+	return r.codec.Encode(context.Background(), agg, typeName)
+}
+
+func (r reportCodec) MedianFromReport(report ocrtypes.Report) (*big.Int, error) {
+	agg := &aggregatedAttributedObservation{}
+	if err := r.codec.Decode(context.Background(), report, agg, typeName); err != nil {
+		return nil, err
+	}
+	medianObservation := len(agg.Observations) / 2
+	return agg.Observations[medianObservation], nil
+}
+
+func (r reportCodec) MaxReportLength(n int) (int, error) {
```

Comment:
Any reason these are value method instead of pointer?
```suggestion
func (r *reportCodec) MaxReportLength(n int) (int, error) {
```
I'm surprised since the constructor returns a pointer.

Reply:
It's likely auto-generated the functions and then forgot to change it.